rskj-core/src/main/java/co/rsk/core/RskFactory.java

```diff
@@ -113,7 +114,9 @@ public Rsk getRsk(WorldManager worldManager,
         }
         if (rskSystemProperties.isBlocksEnabled()) {
             setupRecorder(rsk, rskSystemProperties.blocksRecorder());
-            setupPlayer(rsk, rskSystemProperties.blocksPlayer());
+            rsk.setIsPlayingBlocks(true);
```

Comment:
Check if moving it outside the thread doesn't change behavior.

Reply:
Great finding!